app/src/main/java/com/termux/api/apis/BluetoothScanAPI.java

```diff
@@ -0,0 +1,122 @@
+package com.termux.api.apis;
+
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothDevice;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.util.JsonWriter;
+
+import com.termux.api.util.ResultReturner;
+import com.termux.shared.logger.Logger;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class BluetoothScanAPI {
+
+    private static final ArrayList<BluetoothDevice> devices = new ArrayList<>();
```

Comment:
I think it's strictly necessary to track and return last-seen timestamps. It also looks like the scan API will trigger multiple times for the same devices as it's repeatedly discovered, and a longer scan with the code would result in multiple copies of the same device in this array and the returned output.

This seems like it should be `HashMap<BluetoothDevice, long>`.